### ISSUE TITLE
 [Feature Request] Add support for other git services for autoupdates

### ISSUE THREAD

**Issue by Potajito** [mijorus/gearlever#179]
Currently, we are unable to use another git services (codeberg, gitlab) to autoupdate appimages using the Update management feature, and other similar services could share this feature, so urls like:
https://codeberg.org/sonusmix/sonusmix/releases/download/v0.1.0/org.sonusmix.Sonusmix-0.1.0.AppImage
could be used as:
https://codeberg.org/sonusmix/sonusmix/releases/download/*/org.sonusmix.Sonusmix-*.AppImage
in order to mimic the github behaviour.

**Comment by prontok** [comment]
For codeberg, you can use the zsync
"zsync|https://codeberg.org/sonusmix/sonusmix/releases/download/v0.1.0/org.sonusmix.Sonusmix-0.1.0.AppImage.zsync"

**Comment by dCo3lh0** [comment]
another thing is the possibility to change the default config given by the ZSync files, that sometimes have the link for downloading wrong.

**Comment by paulstenning** [comment]
I have a similar issue with Krita where I normally use the nightly builds here
https://cdn.kde.org/ci-builds/graphics/krita/krita-5.2/linux/

It would be good to be able to use a URL such as
https://cdn.kde.org/ci-builds/graphics/krita/krita-5.2/linux/krita-5.2.9-prealpha-*-x86_64.AppImage

I am using zsync for now, thank you for the suggestion prontok.
Maybe adding zsync support could be a good future option for situations like this?

**Comment by hotlittlewhitedog** [comment]
I agree that support should be extended to other sources as Gitlab because it's impossible to enter the URL and provide updates. You should look at following Android project ([Obtainium](https://github.com/ImranR98/Obtainium)) that supports a lot of sources and also allows to give a regex pattern as you did for Github.

**Comment by mijorus** [comment]
> It would be good to be able to use a URL such as https://cdn.kde.org/ci-builds/graphics/krita/krita-5.2/linux/krita-5.2.9-prealpha-*-x86_64.AppImage


you can't do that, this is not how HTTP works 😅. It's not a filesystem

**Comment by mijorus** [comment]
Btw, I started following the AppImage official documentation which only supports Github, Pling store and  a direct link.
I might consider adding some more sources at this point
https://github.com/AppImage/AppImageSpec/blob/master/draft.md#update-information

**Comment by hotlittlewhitedog** [comment]
Git is not only Github 😸

**Comment by mijorus** [comment]
> Git is not only Github 😸
> […](#)

At the same time, there isn't a "git" way to handle releases 😅

**Comment by ikyquocked** [comment]
A lot of apps are on GitLab, not so many as on GitHub, but I prefer GitLab since it's fully open-source and no MS blobs. GitLab rules

**Comment by NullDott** [comment]
> I have a similar issue with Krita where I normally use the nightly builds here https://cdn.kde.org/ci-builds/graphics/krita/krita-5.2/linux/
> 
> It would be good to be able to use a URL such as https://cdn.kde.org/ci-builds/graphics/krita/krita-5.2/linux/krita-5.2.9-prealpha-*-x86_64.AppImage
> 
> I am using zsync for now, thank you for the suggestion prontok. Maybe adding zsync support could be a good future option for situations like this?

Hi, sorry to bother you, but how did you added the zsync file in your Krita entry? I try to open/add it in the update information section but Gear Lever keeps showing the link red.
